src/GafferSceneUI/ParameterInspector.cpp

```diff
@@ -158,6 +158,13 @@ Gaffer::ValuePlugPtr ParameterInspector::source( const GafferScene::SceneAlgo::H
 		}
 	}
 
+	else
+	{
+		editWarning = boost::str(
+			boost::format( "Edits cannot be made to a %s node" ) % history->scene->node()->typeName()
+		);
+	}
```

Comment:
Perhaps I didn't document this well enough, but the intention with `source()` was that `editWarning` would only be filled when it returned non-null. As in "here is a plug you can edit, but here is a warning you should heed about that".

This is a generic failure case, so I don't think it's ideal to make every different Inspector subclass responsible for it - they're meant to be responsible for the specifics that the base class can't know about. I think we should be able to handle this entirely within the base class somehow. Perhaps when the recursion to `history->predecessors` bottoms out?